### PR TITLE
W-16508382: Remove ClassloadingTestCase and migrate missing tests to ApplicationDeploymentClassloadingTestCase

### PR DESCRIPTION
**Pull request by hyerans**
…ApplicationDeploymentClassloadingTestCase

### CODE REVIEW

File: modules/deployment/src/test/java/org/mule/runtime/module/deployment/test/internal/ApplicationDeploymentClassloadingTestCase.java [hunk 1]
```diff
@@ -137,6 +137,58 @@ public void before() {
...
+    addPackedAppFromBuilder(extendPackagesAppFileBuilder);
+    startDeployment();
+    assertDeploymentSuccess(applicationDeploymentListener, extendPackagesAppFileBuilder.getId());
+    executeApplicationFlow("main");
+  }
+
+  @Test
+  @Issue("W-16508382: Cover missing test cases from ClassloadingTestCase by adding tests for external library dependencies and custom package extensions.")
+  public void canDependOnExternalLibrary() throws Exception {
+    File externalJarFile = new JarCompiler().compiling(getResourceFile("/com/example/external/ExternalClass.java"))
```
Comment (rbourbonnavarro) [inline modules/deployment/src/test/java/org/mule/runtime/module/deployment/test/internal/ApplicationDeploymentClassloadingTestCase.java:178]: The variable name doesn't reflect the content, it's not a JAR but a class

File: modules/deployment/src/test/java/org/mule/runtime/module/deployment/test/internal/ApplicationDeploymentClassloadingTestCase.java [hunk 2]
```diff
@@ -137,6 +137,58 @@ public void before() {
...
+        .compile("extension.jar");
+
+    // Create an application that depends on the external library
+    ApplicationFileBuilder externalLibAppFileBuilder = appFileBuilder("appWithExternalLib")
+        .definedBy("app-with-external-lib-config.xml")
+        .dependingOn(new JarFileBuilder("external-lib", externalJarFile));
```
Comment (rbourbonnavarro) [inline modules/deployment/src/test/java/org/mule/runtime/module/deployment/test/internal/ApplicationDeploymentClassloadingTestCase.java:184]: In the original test the class wasn't in a dependency but in the app itself.

File: modules/deployment/src/test/java/org/mule/runtime/module/deployment/test/internal/ApplicationDeploymentClassloadingTestCase.java [hunk 3]
```diff
@@ -137,6 +137,58 @@ public void before() {
...
+
+    addPackedAppFromBuilder(externalLibAppFileBuilder);
+    startDeployment();
+    assertDeploymentSuccess(applicationDeploymentListener, externalLibAppFileBuilder.getId());
+    executeApplicationFlow("main");
```
Comment (rbourbonnavarro) [inline modules/deployment/src/test/java/org/mule/runtime/module/deployment/test/internal/ApplicationDeploymentClassloadingTestCase.java:189]: The result of the flow execution should be checked.

**Review by rbourbonnavarro — APPROVED** [review]

File: modules/deployment/src/test/java/org/mule/runtime/module/deployment/test/internal/ApplicationDeploymentClassloadingTestCase.java (hunk 3 above)
Comment (rbourbonnavarro) [inline modules/deployment/src/test/java/org/mule/runtime/module/deployment/test/internal/ApplicationDeploymentClassloadingTestCase.java:189]: The result of the flow execution should be checked.

**Review by rbourbonnavarro — CHANGES_REQUESTED** [review]

File: modules/deployment/src/test/java/org/mule/runtime/module/deployment/test/internal/ApplicationDeploymentClassloadingTestCase.java (hunk 1 above)
Comment (hyerans) [reply modules/deployment/src/test/java/org/mule/runtime/module/deployment/test/internal/ApplicationDeploymentClassloadingTestCase.java]: removed this test, and added canDependOnXerces test

**Pull request closed without merging**